google/cloud/spanner/internal/defaults.cc

```diff
@@ -105,6 +105,20 @@ Options DefaultOptions(Options opts) {
       (std::min)(min_sessions, max_sessions_per_channel * num_channels);
 
   if (!opts.has<spanner::RouteToLeaderOption>()) {
+#if 1
+    // TODO(#11111): Enable on-by-default behavior.
+    opts.set<spanner::RouteToLeaderOption>(false);
+    if (auto e = internal::GetEnv("GOOGLE_CLOUD_CPP_SPANNER_ROUTE_TO_LEADER")) {
```

Comment:
You have previously argued that environment variables should override the values set in the code.  This does not seem to do that, thoughts?

Reply:
You're correct that this environment variable is different ... it is for overriding the default value rather than a value set in code.  Let me think about that and get back to you on whether that difference is a feature or a bug.  Thanks.

Reply:
Ac. SGTM.  We may want to consider some updates to our ADR regarding this, but that can happen elsewhere.